include/onnxruntime/core/common/logging/logging.h

```diff
@@ -333,5 +345,17 @@ unsigned int GetThreadId();
 */
 unsigned int GetProcessId();
 
+/**
+   If the ONNXRuntimeTraceLoggingProvider ETW Provider is enabled, then adds to the existing logger.
+*/
+std::unique_ptr<ISink> EnhanceLoggerWithEtw(std::unique_ptr<ISink> existingLogger, logging::Severity originalSeverity,
```

Comment:
Could this be defined in a file in platform/windows/logging instead of the base logging.h given it's Windows specific?

Reply:
It could but prefer to keep as-is today. 

My reasoning is 
a) for current implementation - ease of use of accessing it including on non-Windows essentially a no-op 
b) This dynamic logging support with TraceLogging need not be limited to only Windows in the future. TraceLogging is actually partially cross-platform and available on [Linux](https://github.com/microsoft/tracelogging) so this doesn't have to be Windows ETW specific. However, it's outside scope and complexity for this PR to enable this